xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -523,17 +528,23 @@ void CGUIFontTTF::DrawTextInternal(float x,
       else if (maxPixelWidth > 0 && cursorX > maxPixelWidth)
         break;  // exceeded max allowed width - stop rendering
 
-      RenderCharacter(startX + cursorX, startY, ch, color, !scrolling, *tempVertices);
+      auto currentGlyph = glyphPos.front();
```

Comment:
it's a queue because there are two loops, the first one getting the characters and save them in the queue.
but the second loop will render the characters from the back, after submitting the code I had an idea to loop the glyphs backwards instead of forward in the second loop but I don't know if this will work.

Reply:
I think I dont really understand. A `queue` is `first in first out` right? Then how come they render in second loop backwards?

Reply:
You are right, we don't need the queue, when I was debugging at first the characters weren't synched because I made glyphs info in the cached characters which gave me wrong values, so I thought that the rendering was backward. Anyways I fixed it.